lib/ansible/runner.rb

```diff
@@ -354,6 +358,36 @@ def env_dir(base_dir)
         FileUtils.mkdir_p(File.join(base_dir, "env")).first
       end
 
+      def pid_file(base_dir)
+        File.join(base_dir, "pid")
+      end
+
+      def wait_for(path)
+        require "listen"
+        require "concurrent"
+
+        dir_name  = File.dirname(path)
+        file_name = File.basename(path)
+
+        path_created = Concurrent::Event.new
+
+        listener = Listen.to(dir_name, :only => %r{\A#{file_name}\z}) do |modified, added, _removed|
+          path_created.set if added.include?(path) || modified.include?(path)
+        end
+
+        thread = Thread.new { listener.start }
+
+        begin
+          res = yield
+          path_created.wait
```

Comment:
Should we add some reasonable timeout here like 5 seconds?

Reply:
or maybe make it a parameter to `wait_for`?

Reply:
:+1: yeah sounds good